dockerspawner/dockerspawner.py

```diff
@@ -398,6 +412,27 @@ def start(self, image=None, extra_create_kwargs=None,
                 volumes=self.volume_mount_points,
                 name=self.container_name,
             )
+
+            self.upgrade_kwags(self.extra_create_kwargs)
+
+            #https://github.com/docker/docker-py/issues/982
+            if "networkalias" in self.extra_create_kwargs:
```

Comment:
I don't think we should support custom things like this. `extra_create_kwargs` should be passed to `create` exactly as it is. The issue linked in the above comment explains how to write the correct networking_config according to the docker API docs.

Reply:
I understand. But there's already an exception in the code for the mem_limit and port_bindings.
Also, i needed to add the apply_templates and this is only available inside your code.

The whole goal of this change is to create subdomains to each user. this is the core of the change